engine/src/main/java/org/destinationsol/game/SolGame.java

```diff
@@ -126,113 +161,49 @@
     private RespawnState respawnState;
     private SortedMap<Integer, List<UpdateAwareSystem>> onPausedUpdateSystems;
     private SortedMap<Integer, List<UpdateAwareSystem>> updateSystems;
-    private Context context;
 
     private EntitySystemManager entitySystemManager;
+    private final MainGameScreen mainGameScreen;
 
-    public SolGame(String shipName, boolean isTutorial, boolean isNewGame, CommonDrawer commonDrawer, Context context,
-                   WorldConfig worldConfig) {
-        // TODO: make this non-static
-        FactionInfo.init();
-
-        this.context = context;
-
-        this.isTutorial = isTutorial;
-        solApplication = context.get(SolApplication.class);
-
-        //TODO this no longer needs to be instantiated in SolGame
-        GameColors gameColors = new GameColors();
-        context.put(GameColors.class, gameColors);
-
-        //TODO this no longer needs to be instantiated in SolGame
-        soundManager = solApplication.getSoundManager();
-        SpecialSounds specialSounds = new SpecialSounds(soundManager);
-        context.put(SpecialSounds.class, specialSounds);
-
-        //TODO these no longer need to be instantiated in SolGame
-        GameDrawer drawer = new GameDrawer(commonDrawer);
-        context.put(GameDrawer.class, drawer);
-        DrawableManager drawableManager = new DrawableManager(drawer);
-        context.put(DrawableManager.class, drawableManager);
-
-        //TODO refactor the usages of this to be independent from SolGame, so that this can be deleted
-        // The only methods still using this are those that require changing an interface signature (UpdateAwareSystem, SolObject, and Drawable)
-        SolCam camera = context.get(SolCam.class);
-        solCamReference = camera;
-
-        gameScreens = new GameScreens(solApplication, context);
+    public MainGameScreen getMainGameScreen() {
+        return mainGameScreen;
+    }
 
+    @Inject
+    public SolGame(SolApplication solApplication, PlanetConfigManager planetConfigManager, MazeConfigManager mazeConfigManager, BeltConfigManager beltConfigManager) {
+        FactionInfo.init();
+        this.solApplication = solApplication;
         boolean isMobile = solApplication.isMobile();
+
         if (!isMobile) {
             mainGameScreen = (MainGameScreen) Assets.getAssetHelper().get(new ResourceUrn(NUI_MAIN_GAME_SCREEN_DESKTOP_URI), UIElement.class).get().getRootWidget();
         } else {
             mainGameScreen = (MainGameScreen) Assets.getAssetHelper().get(new ResourceUrn(NUI_MAIN_GAME_SCREEN_MOBILE_URI), UIElement.class).get().getRootWidget();
         }
+        // TODO: resolve hack to set solApplication
+        mainGameScreen.setSolApplication(solApplication);
```

Comment:
If you're still planning to resolve this, then you can start by looking at the injection code in `NuiManager#pushScreen`.

Reply:
hmm, how would this work in practice, if we push a screen it should be in the context it was instantiated in :?.

Reply:
The existing behaviour is that the screen is re-injected when pushed onto the UI stack (displayed). It should probably use the currently active context (e.g. `appContext` for the main menu and `gameContext` for in-game).
https://github.com/MovingBlocks/DestinationSol/blob/30700e93270716c5e57fff16784f87153a97b9f3/engine/src/main/java/org/destinationsol/ui/nui/NUIManager.java#L287-L295
(I think that replacing `InjectionHelper#inject` with `BeanContext#inject` would probably be sufficient)

#631 re-works the logic in `NUIManager` to inject the screens upon creation, rather than upon display, so then they would then inherit the context they were instantiated in. For now though, the existing behaviour should be preserved.